lib/Dialect/HW/HWOps.cpp

```diff
@@ -2147,6 +2155,18 @@ LogicalResult StructExtractOp::canonicalize(StructExtractOp op,
   return failure();
 }
 
+void StructExtractOp::getAsmResultNames(
+    function_ref<void(Value, StringRef)> setNameFn) {
+  auto structType = type_cast<StructType>(getInput().getType());
+  for (auto field : structType.getElements()) {
+    if (field.name == getField()) {
+      setNameFn(getResult(), field.name.str());
+      return;
+    }
+  }
+  llvm_unreachable("field not found");
```

Comment:
Plz remove. This function should never assert or crap out since it could be called on bad IR which is being debugged.